src/libraries/Native/Unix/System.Native/pal_autoreleasepool.m

```diff
@@ -42,3 +42,9 @@ void SystemNative_DrainAutoreleasePool(void* pool)
 {
     [((NSAutoreleasePool*)pool) drain];
 }
+
+void SystemNative_AllocateThreadlocalAutoreleasePool(void)
+{
+    EnsureNSThreadIsMultiThreaded();
+    (void)SystemNative_CreateAutoreleasePool();
```

Comment:
Is the created NSAutoreleasePool object guaranteed to be drained at thread exit?

Reply:
Yes. From https://developer.apple.com/documentation/foundation/nsautoreleasepool#1651513.

>Each thread (including the main thread) maintains its own stack of NSAutoreleasePool objects (see Threads). As new pools are created, they get added to the top of the stack. When pools are deallocated, they are removed from the stack. Autoreleased objects are placed into the top autorelease pool for the current thread. When a thread terminates, it automatically drains all of the autorelease pools associated with itself.